src/baldr/graphtile.cc

```diff
@@ -980,38 +980,28 @@ const TransitDeparture* GraphTile::GetNextDeparture(const uint32_t lineid,
   // Iterate through departures until one is found with valid date, dow or
   // calendar date, and does not have a calendar exception.
   for (; found < count && departures_[found].lineid() == lineid; ++found) {
-    // Make sure valid departure time for the exact/fixed schedule
-    if (departures_[found].type() == kFixedSchedule) {
-      if (departures_[found].departure_time() >= current_time &&
-          GetTransitSchedule(departures_[found].schedule_index())
-              ->IsValid(day, dow, date_before_tile) &&
-          (!wheelchair || departures_[found].wheelchair_accessible()) &&
-          (!bicycle || departures_[found].bicycle_accessible())) {
-        return &departures_[found];
-      }
+    // Make sure it falls within the schedule and departure props are valid
+    const auto& d = departures_[found];
+    if (!GetTransitSchedule(d.schedule_index())->IsValid(day, dow, date_before_tile) ||
+        (wheelchair && !d.wheelchair_accessible()) || (bicycle && !d.bicycle_accessible())) {
+      continue;
+    }
```

Comment:
yeah, it's a refactor, but I find it much more legible. there's no functional changes and this stuff is fairly well tested